Teaching/Miscellaneous/MATMUL.md

```diff
@@ -0,0 +1,59 @@
+# MATMUL Optimisation
+## Background
+Matrix multiplication is a fundamental operation in numerous computational and mathematical tasks. This operation is fundamental in many areas, including graphics rendering, scientific simulations, machine learning, and more. Given the significance of matrix multiplication these various fields, failing to implement it with performance in mind can pose a severe computational bottleneck.
+
+A naïve implementation of this algorithm (shown below) exhibits a time complexity of O(n^3) for two n × n matrices. 
+``` python
+import sys, random
```

Comment:
No need to give the entire code here. You can link to a code perhaps, or just explain it.

Reply:
changed it to a gist link